// @parse/simple-mailgun-adapter/index.js

// var Mailgun = require('mailgun-js');

// var SimpleMailgunAdapter = mailgunOptions => {
//   if (!mailgunOptions || !mailgunOptions.apiKey || !mailgunOptions.domain || !mailgunOptions.fromAddress) {
//     throw 'SimpleMailgunAdapter requires an API Key, domain, and fromAddress.';
//   }
//   var mailgun = Mailgun(mailgunOptions);

//   var sendMail = mail => {
//     var data = Object.assign({}, mail, { from: mailgunOptions.fromAddress });

//     return new Promise((resolve, reject) => {
//       mailgun.messages().send(data, (err, body) => {
//         if (typeof err !== 'undefined') {
//           reject(err);
//           return;
//         }
//         resolve(body);
//       });
//     });
//   }

//   return Object.freeze({
//     sendMail: sendMail
//   });
// }

// module.exports = SimpleMailgunAdapter


var api = new ParseServer({
  databaseURI: databaseUri || 'mongodb://localhost:27017/dev',
  cloud: process.env.CLOUD_CODE_MAIN || __dirname + '/cloud/main.js',
  appId: process.env.APP_ID || 'myAppId',
  appName: process.env.APP_NAME || 'myApp',
  masterKey: process.env.MASTER_KEY || 'myMasterKey', //Add your master key here. Keep it secret!
  serverURL: process.env.SERVER_URL || 'http://localhost:1337/parse',  // Don't forget to change to https if needed
  publicServerURL: process.env.SERVER_URL || 'http://localhost:1337/parse',
  liveQuery: {
    classNames: ["Posts", "Comments"] // List of classes to support for query subscriptions
  },
  emailAdapter: {
    module: '@parse/simple-mailgun-adapter',
    options: {
      fromAddress: 'Name <your@email.com>',
      domain: process.env.MAILGUN_DOMAIN,
      apiKey: process.env.MAILGUN_API_KEY,
    }
  }
});
